consensus/src/epoch_manager.rs

```diff
@@ -286,6 +292,9 @@ impl<T: Payload> EpochManager<T> {
 
         info!("Update SafetyRules");
 
+        // @REVIEW: the "safety rules" system also has data that's required for liveness. Seems like it'd be nice
+        // to be able to recover just based on the consensus db / storage systems (at least if an operator chooses)
```

Comment:
how to manage the fault tolerance between tcb and consensus is not clear yet. although it's probably not something like this (we don't do nice-to-have thing right now)